### ISSUE TITLE
Unable to modify rolls to hide Dice So Nice rolls from other players

### ISSUE THREAD

**Issue by cs96and** [RedReign/FoundryVTT-BetterRolls5e#296]
Hi, I'm the current maintainer of the [Cautious Gamemaster's Pack](https://github.com/cs96and/FoundryVTT-CGMP).  One of the features of that module is the ability to hide rolls made by hidden tokens.  This mostly works fine (apart from #295), but when Better Rolls is enabled, the Dice So Nice animations still display on other players' screens.

I tried catching the `messageBetterRolls` hook to modify the whisper data, but this whisper data is not passed to the Dice So Nice's `showForRoll()` call

#### `CustomItemRoll.toMessage()` (custom-roll.js line 837)...
```javascript
await Hooks.callAll("messageBetterRolls", this, chatData);
await this.dicePool.flush(hasMaestroSound);
```

My hook callback is modifying the chatData so the `whisper` property is updated, but this is not passed to `dicePool.flush()`. 

#### `DiceCollection.flush()` (dice-collection.js line 52)...
```javascript
const hasDice = pool.dice.length > 0;
if (game.dice3d && hasDice) {
		const wd = Utils.getWhisperData();
		await game.dice3d.showForRoll(pool, game.user, true, wd.whisper, wd.blind || false);
}
```

It just calls `Utils.getWhisperData()` to get the whisper data from the roll mode drop down.  However the whisper data was already calculated in `CustomItemRoll.toMessage()` and then possibly modified.

TL;DR - it would be nice if `DiceCollection.flush()` optionally used the already calculated whisper data, rather then calculating it again from scratch.

**Comment by CarlosFdez** [comment]
Makes sense to me for it to do so. I'll add whisper data as an optional parameter to flush and it should work out just fine.
